### DISCUSSION TITLE
Need help with Partitioning config

### DISCUSSION THREAD

**Discussion by quangbahoa** [stuttter/ludicrousdb#162] [Help] [answered]
Hi there,

I'm try to setup db partitioning for multisite with 15 child sites. Currently I'm config as `Sample Configuration 2: Partitioning` and it work. All child site tables are stored on `blogdb`. Now I want config 5 child sites on a db. Eg,
 - Site id 2-5 stored on db name call `child_db_2_5`
 - Site id 6-10 stored on db name call `child_db_6_10`
 - Site id 11-15 stored on db name call `child_db_11_15`

How can I do that?

Thanks for your help,
Regards,

**Comment by deepakchandarr** [comment]
even i would like to know if there is a solution

**Comment by JJJ** [comment, marked as answer]
Yes. Route on the numeric blog ID parsed from `$wpdb->table`, and return the dataset for the matching range:

```php
$wpdb->add_callback(
	function ( $query, $wpdb ) {
		$pattern = '/^' . preg_quote( $wpdb->base_prefix, '/' ) . '(\d+)_/';

		if ( ! preg_match( $pattern, $wpdb->table, $matches ) ) {
			return null;
		}

		$blog_id = (int) $matches[1];

		if ( $blog_id >= 2 && $blog_id <= 5 ) {
			return 'child_db_2_5';
		}

		if ( $blog_id <= 10 ) {
			return 'child_db_6_10';
		}

		if ( $blog_id <= 15 ) {
			return 'child_db_11_15';
		}

		return null;
	}
);
```

Add each returned dataset with `$wpdb->add_database()` first. The site's tables must already exist in that physical database, and related tables such as posts and post metadata should move together. Returning `null` lets the normal/default routing handle the main site, network-global tables, and sites outside these ranges.

The same approach is now documented in the wiki's Partitioning Recipes page.
